website/mailing_list/views.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+from flask import request
+
+from website.project.decorators import (
+    must_be_valid_project,
+    must_have_permission,
+    must_not_be_registration,
+)
+from website.util.permissions import ADMIN, READ
+from website.mailing_list import utils
+
+
+###############################################################################
+# View Functions
+###############################################################################
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def get_node_mailing_list(node, auth, **kwargs):
+    return format_node_data_recursive([node], auth.user)
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def enable_mailing_list(node, **kwargs):
+    utils.celery_create_list(node._id)
+    node.mailing_enabled = True
+    node.save()
+    return {'message': 'Successfully enabled mailing lists', 'success': True}
+
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_not_be_registration
+def disable_mailing_list(node, **kwargs):
+    utils.celery_delete_list(node._id)
+    node.mailing_updated = True
+    node.mailing_enabled = False
+    node.save()
+
+def flask_unsubscribe_user(*args, **kwargs):
+    message = request.form
+    utils.unsubscribe_user_hook(message)
```

Comment:
This is a step forward, but can still be improved a bit: The utility function should operate on data without depending on one framework (so we can support flask + django endpoints in the future). But there is no guarantee that another framework would create a `request.form` object that looks or acts the same way.

So to keep this framework agnostic, the utility function call signature should be something like `def unsubscribe_hook(recipient, message, headers)`- the view function is responsible for unpacking the form into data, and the utility function shouldn't depend on the internal structure of special flask objects at all.